dissect/target/plugins/os/windows/tasks.py

```diff
@@ -186,23 +356,331 @@ def strip_namespace(data):
+
+            if trigger_type == "WEEKLY":
+                interval = trigger.trigger_specific0
+
+                # Find weekdays corresponding with trigger specific flags
+                days_of_week = get_days_of_week(trigger.trigger_specific1)
+                unused = [trigger.trigger_specific2]
+
+                record = WeeklyTriggerRecord(
+                    weeks_between_triggers=interval,
+                    days_of_week=days_of_week,
+                    unused=unused,
+                    padding=padding,
+                    reserved2=reserved2,
+                    reserved3=reserved3,
+                )
+
+                yield GroupedRecord("filesystem/windows/task/WeeklyTrigger", [base, record])
+
+            if trigger_type == "MONTHLYDATE":
+                # Convert trigger_specific fields to binary, remove the "0b" prefix, and pad with zeroes to 16 digits
+                day_flag_part1 = bin(trigger.trigger_specific0)[2:].zfill(16)
+                day_flag_part2 = bin(trigger.trigger_specific1)[2:].zfill(16)
+                # Concatenate the two binary strings to form the complete day_flag
+                day_of_month_flag = day_flag_part2 + day_flag_part1
+                # Reverse the day_flag string (LE), find the index of "1", add 1 to get the day of the month
+                day_of_month = day_of_month_flag[::-1].index("1") + 1
+
+                # Find months corresponding with trigger specific flags
+                months_of_year = get_months_of_year(trigger.trigger_specific2)
+
+                record = MonthlyDateTriggerRecord(
+                    day_of_month=day_of_month,
+                    months_of_year=months_of_year,
+                    padding=padding,
+                    reserved2=reserved2,
+                    reserved3=reserved3,
+                )
+
+                yield GroupedRecord("filesystem/windows/task/MonthlyDateTrigger", [base, record])
+
+            if trigger_type == "MONTHLYDOW":
+                week = trigger.trigger_specific0
+                week_strings = [
+                    "FIRST_WEEK",
+                    "SECOND_WEEK",
+                    "THIRD_WEEK",
+                    "FOURTH_WEEK",
+                    "LAST_WEEK",
+                ]
+                week = week_strings[week - 1]
+                day = get_days_of_week(trigger.trigger_specific1)
+                months = get_months_of_year(trigger.trigger_specific2)
+                record = MonthlyDowTriggerRecord(
+                    which_week=week,
+                    day_of_week=day,
+                    months_of_year=months,
+                    padding=padding,
+                    reserved2=reserved2,
+                    reserved3=reserved3,
+                )
+
+                yield GroupedRecord("filesystem/windows/task/MonthlyDowTrigger", [base, record])
+
+
+class XmlTask:
+    def __init__(self, xml_data) -> None:
```

Comment:
```suggestion
    def __init__(self, xml_data):
```
We don't generally add the type hint to `__init__`. Also missing type hint for `xml_data`.